### PR TITLE
DVL model updates and demo

### PR DESCRIPTION
**Pull request by dtdavi1**
Added all of the DVL models from the "dave" repository into this one and updated everything so that it runs--same launch commands as in the tutorial, but from the nps_uw_sensors_gazebo package (I'll update the wiki tutorial in the next couple of days).

Added a "side-by-side" demo of the 3 meshes that we've got so far.  The demo can be launched with the command "roslaunch nps_uw_sensors_gazebo whoi_dvl_static_side_by_side.launch" and will launch both a Gazebo window and an RVIZ window (the RVIZ window has better lighting).  The plugins are "live" for all 3 models, so the DVL topics are being published.  The models are all attached to the same base link (easier to get into RVIZ), so moving one will probably move all 3.

### CODE REVIEW

**Comment by bsb808** [comment]
Cool - does that mean we should close [Dave PR #19](https://github.com/Field-Robotics-Lab/dave/pull/19)?

**Comment by dtdavi1** [comment]
> Cool - does that mean we should close [Dave PR #19](https://github.com/Field-Robotics-Lab/dave/pull/19)?

I'd leave it for now, but don't accept it.  I don't see any reason to maintain the Dave versions any longer (too easy to get mismatches that cause trouble), so as soon as this is merged I'll delete all of the DVL stuff from that one.  Once the deletes are committed the merge will close out that thread.

**Comment by bsb808** [comment]
I'll take a look now.  I'm working on the highlights and want to include the DVL models.

File: models/uuvsim_teledyne_whn_urdf/model.config [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<model>
+  <name>teledyne_whn</name>
```
Comment (bsb808) [inline models/uuvsim_teledyne_whn_urdf/model.config]: All of the WHN models have the same name, which makes it a bit confusing.

**Review by bsb808 — COMMENTED** [review]
I would recommend creating standalone Gazebo models for each of the DVL models.   This allows us to have the mesh and model.sdf and model.config files in a labeled directory under models.

As an example, see the 3d_lidar model here: https://github.com/osrf/vrx/tree/master/vrx_gazebo/models
The associated `model.sdf` file includes the visual, collision and inertial values.

Then you can re-use the mesh in the xacro or urdf that defines any mounting, joints, plugins, etc.  See the example here: https://github.com/osrf/vrx/blob/master/wamv_gazebo/urdf/sensors/wamv_3d_lidar.xacro

**Comment by BruceMty** [comment]
It looks like some of the .xacro files in urdf are about the same except for some physics properties, for example uuvsim_nortek_dvl_1000_300.xacro and uuvsim_nortek_dvl_100_4000.xacro.  Would it be reasonable to define macros for each of these configurations that wrap a common macro?  Or maybe being separate helps pick-and-choose?

File: urdf/uuvsim_teledyne_whn_robot.xacro [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0"?>
```
Comment (BruceMty) [inline urdf/uuvsim_teledyne_whn_robot.xacro:1]: Some of these .xacro files have their execute bit set.

File: models/uuvsim_teledyne_whn_urdf/model.config (hunk 1 above)
Comment (dtdavi1) [reply models/uuvsim_teledyne_whn_urdf/model.config]: Fixed in latest commit (I think).

File: urdf/uuvsim_teledyne_whn_robot.xacro (hunk 2 above)
Comment (dtdavi1) [reply urdf/uuvsim_teledyne_whn_robot.xacro:1]: Fixed in latest commit.

**Comment by dtdavi1** [comment]
> I would recommend creating standalone Gazebo models for each of the DVL models. This allows us to have the mesh and model.sdf and model.config files in a labeled directory under models.
> 
> As an example, see the 3d_lidar model here: https://github.com/osrf/vrx/tree/master/vrx_gazebo/models
> The associated `model.sdf` file includes the visual, collision and inertial values.
> 
> Then you can re-use the mesh in the xacro or urdf that defines any mounting, joints, plugins, etc. See the example here: https://github.com/osrf/vrx/blob/master/wamv_gazebo/urdf/sensors/wamv_3d_lidar.xacro

Created SDF models for all of the performer DVLs.  Modified all of the Xacro files to use the SDF model meshes.  See the [commit](https://github.com/Field-Robotics-Lab/nps_uw_sensors_gazebo/pull/6/commits/1352f18d2d8b00760670acee446ce7640eccbbf1) message for a more thorough description.

**Comment by dtdavi1** [comment]
>> It looks like some of the .xacro files in urdf are about the same except for some physics properties, for example uuvsim_nortek_dvl_1000_300.xacro and uuvsim_nortek_dvl_100_4000.xacro. Would it be reasonable to define macros for each of these configurations that wrap a common macro? Or maybe being separate helps pick-and-choose?

>Probably.  There are a lot of things that can probably be refactored or simplified.  It's probably too big a change for release 1, so it will need to go onto the to-do list.

Okay, good.

File: models/uuvsim_teledyne_whn_urdf/model.config (hunk 1 above)
Comment (BruceMty) [reply models/uuvsim_teledyne_whn_urdf/model.config]: Looks good.

File: urdf/uuvsim_teledyne_whn_robot.xacro (hunk 2 above)
Comment (BruceMty) [reply urdf/uuvsim_teledyne_whn_robot.xacro:1]: Done.

**Comment by BruceMty** [comment]
I'm good with this PR.

**Pull request merged**